### DISCUSSION TITLE
Страница "Ядро (Core)"

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#94] [Страницы Wiki]
Предложения по улучшения странички http://fhir-ru.zendoc.me/core предлагаю писать в этой дискуссии.

**Comment by Vrodvi** [comment]
Предложения:

1) Разобраться почему статусы профилей отображаются в полном разнобое:
![image](https://user-images.githubusercontent.com/92928137/212002106-5c155a01-ecc6-41b2-b65b-a1cdda85ad9e.png)

1.1) Перевести все статусы на русский язык и создать страницу\раздел с легендой, описывающей значение каждого статуса и схемой статусных переходов.

1.1.1) Схему статусных переходов оформить в текстовом формате (https://plantuml.com/state-diagram или https://mermaid.js.org/syntax/stateDiagram.html) для удобства контроля изменений в версиях, с динамическим формированием графического представления.

1.2) Исправить атрибут статус во всех перечисленных профилях на единообразные значения с учётом вышеописанных замечаний (статусы должны отображаться для всех профилей, статусы должны отображаться единообразно, статусы должны быть на русском языке)

**Comment by Vrodvi** [comment]
Предложения:

2. Разложить профили по тематическим разделам, как это делается во всех зарубежных руководствах, формируемых стандартным инструментом FHIR Publisher:
![image](https://user-images.githubusercontent.com/92928137/212005406-04b027fe-d154-4fc6-a3fa-6962927a5aed.png)

2.1) Создать разделы и разложить по ним профили (желательно чтобы это осуществлялось автоматически, как реализовано в стандартном инструменте FHIR Publisher):
- Профили основных ресурсов:
  - тип ресурса
    - конкретный профиль
      - унаследованный профиль
- Расширения:
  - конкретное расширение
- Терминологии:
  - системы кодирования
    - конкретная система кодирования
  - наборы значений
    -  конкретный набор значений
  - маппирования между наборами значений
    -  конкретное маппирование

2.2) Добавить оглавление с перечислением разделов

**Comment by Vrodvi** [comment]
Предложения:

3. Перевести заголовки на русский язык
![image](https://user-images.githubusercontent.com/92928137/212006943-2ff78fd5-522a-449f-b04e-45296cc6d42d.png)

**Comment by Vrodvi** [comment]
По поводу описания http://fhir.ru/core#:fsh - допущена ошибка 
![image](https://github.com/fhir-ru/core/assets/92928137/82b64197-ea8f-48cf-ae62-d56ea799fb8c)
- в правиле подбора кодов для набора значений указано взятие кодов из набора значений, что не правильно.
Набор значений может содержать только указание кодов из справочника (ресурс CodeSystem).
Конкретно для пола пациента - должен быть указание на взятие кодов из федерального справочника ФНСИ ([1.2.643.5.1.13.13.11.1040](https://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.13.11.1040)).

Кстати на исходной странице (http://fhir.ru/core.valuesets.core-vs-nsi-gender) правило подбора кодов также неправильно описано - там указан идентификатор "системы идентификации" (NamingSystem, при этом ещё и сам идентификатор неправильно написан - `http://fhir.ru/core/systems/nsi-gender` против `http://fhir.ru/core/cs/core-cs-nsi-gender` (как говорится, если допустить 4 ошибки в слове хлеб, то получится пиво)), а не идентификатор справочника.
![image](https://github.com/fhir-ru/core/assets/92928137/252bf79e-d231-4a91-96a8-04141f50a905)

Подробнее по проблемам с описаниями справочников можно посмотреть тут - https://github.com/fhir-ru/core/discussions/220.

**Comment by GorbachevaT** [comment]
Добрый вечер, коллеги!
Выложила описание  "Пациент и пол пациента (версия 2)", там же, где  была версия 1 -  http://fhir.ru/core#:fsh
Прошу подтверждения что двигаюсь в верном направлении.

**Reply by Vrodvi** [reply]
@eugenekogan-spb @GorbachevaT 

# 1) Вставлен неописанный профиль адреса.
```
Profile:     CoreAddress
Id:          core-address
Parent:      Address
Title:       "Адресс (Ru Core)"
Description: "Адрес с кодом адреса и кодом дома по ФИАС"

// TODO
```
- необходимо описать в формате FSH профиль адреса, взяв за основу описание со страницы http://fhir.ru/core.datatypes.Address
(не забыть также описать два расширения которые там используются и выяснить что за набор значений (ValueSet) представлен на указанной странице)

# 2) Примеры (экземпляры core-organization-instance-1, core-organization-instance-2) для ресурса-организация даны бессмысленные, с незначащим наполнением, что снижает их ценность. 
- Необходимо дать примеры реальных медицинских организаций РФ.

# 3) В профиле пациента атрибут `* name.use 0..1` дан без описания и без каких-либо ограничений - в таком виде причина его добавления не понятна.
- Необходимо либо убрать этот атрибут либо добавить для него описание.
 
# 4) В профиле пациента указание URL профиля дано посреди описания структуры. URL - это часть метаданных профиля и должен быть указан в самом начале профиля, сразу после заголовка.
- Необходимо перенести указание URL к блоку метаданных профиля.

# 5) Непоследовательно используются ссылки между артефактами - в одном месте ссылка дана по url, в другом - по алиасу. Помимо этого в FSH возможны ещё и ссылки по "названиям" артефактов (именами считаются первые строчки в описании артефакта, например - `Profile: CorePatient` или `CodeSystem: CoreCsRosminzdravGender`)
Пример непоследовательных ссылок:
ссылка по url:
```
* gender from http://fhir.ru/core/vs/core-vs-administrative-gender (required)
```
ссылка по алиасу:
```
* include codes from system $CoreCsRosminzdravGender
```
- @eugenekogan-spb  необходимо рассмотреть все обозначенные варианты ссылок между артефактами (см. https://build.fhir.org/ig/HL7/fhir-shorthand/reference.html#references, https://build.fhir.org/ig/HL7/fhir-shorthand/reference.html#referring-to-items), подготовить демонстрацию использования каждого варианта (с обозначением преимуществ/недостатков каждого варианта) и предложить сообществу использование какого-то одного варианта ссылок для единоообразия.

# 6) Для профиля пациент не представлены примеры экземпляров.
- необходимо добавить примеры экземпляров для профиля пациент с реалистичным наполнением.

# 7) В профиле пациента осуществлена попытка переопределить стандартный набор значений, определённый FHIR, который помечен как required (т.е. обязательный к применению и не подлежащий замене)
`* gender from http://fhir.ru/core/vs/core-vs-administrative-gender (required)`
![image](https://github.com/fhir-ru/core/assets/92928137/e6b720b7-ffce-432d-b155-7c6af39a6f8c)
- необходимо исправить ошибочное переопределение стандартного набора значений и использовать альтернативные варианты кодирования пола по российскому справочнику - к примеру, через расширение, или через отдельно определённый маппинг (ConceptMap), или через создание "дополнения" к стандартному справочнику FHIR (см. http://hl7.org/fhir/R4/codesystem.html#supplements)
- необходимо проверить соответствие решений принятых тут https://github.com/fhir-ru/core/issues/12 с реальным описанием кодирования пола пациента на Zendoc.

**Reply by Vrodvi** [reply]
@eugenekogan-spb @GorbachevaT 
По вопросу 7) (пол пациента) также необходимо учесть решение из обсуждения https://github.com/fhir-ru/core/discussions/34, а также учесть наличие ограничения на стандартный FHIR-набор значений для пола пациента, описанные тут - https://fhir-ru.zendoc.me/core.valuesets.vs-administrative-gender.
